app/components/Form/TimePicker.js

```diff
@@ -27,57 +27,98 @@ type Props = {
 };
 
 type State = {
-  value: moment
+  value: moment,
+  tempValue: {
+    hour: string,
+    minute: string
+  }
 };
 
 function parseValue(value) {
   return value ? moment(value) : moment();
 }
 
 class TimePicker extends Component<Props, State> {
+  constructor(props: Props) {
```

Comment:
Is there a specific reason that made you re-add the `constructor`?

Reply:
This was my mistake. I read somewhere that you should only use the `state = {...}` way when the state would not depend on initial props. Like the stackoverflow example below.
**Use constructor when you want to save props data into state**
```js
export default class LoginScreen extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      loading: props.loading,
      loggedIn: props.loggedIn,
    }
  }
}
```
**Otherwise you can directly set the state for hard coded data**
```js
export default class LoginScreen extends React.Component {
  state = {
    loading: false,
    loggedIn: false,
  }
}
```

But this guy says its not the case, so idk https://hackernoon.com/the-constructor-is-dead-long-live-the-constructor-c10871bea599

Reply:
Aha, that makes sense. Guess we can just skip using the constructor then.
Nice article by `Donavon West` btw.!

Reply:
The reason for using constructor was for creating the variable `const value = parseValue(this.props.value);` so that I didn't have to create the same variable three times and make sure that the variable was created before the state was initially set. I can try to do a workaround =)